src/zenml/integrations/aws/step_operators/sagemaker_step_operator.py

```diff
@@ -179,9 +181,14 @@ def launch(
             image_name, self.config.role, **estimator_args
         )
 
+        # SageMaker allows 63 characters at maximum for job name - ZenML uses 60 for safety margin.
+        step_name = Client().get_run_step(info.step_run_id).name
+        training_job_name = f"{info.pipeline.name}-{step_name}"[:55]
+        suffix = "".join(random.choices(string.ascii_lowercase, k=4))
+        unique_training_job_name = f"{training_job_name}-{suffix}"
+
         # Sagemaker doesn't allow any underscores in job/experiment/trial names
-        job_name = f"{info.run_name}-{info.pipeline_step_name}"
-        job_name = job_name.replace("_", "-")
+        sanitized_training_job_name = unique_training_job_name.replace("_", "-")
```

Comment:
I'm thinking this could be abstracted out of the main `launch` functionality as a util outside the integration, which would have the advantage of allowing for a unit test to be written as well. Maybe to live in `src/zenml/utils/string_utils.py`? WDYT? I imagine that there will be other places where we'll hit these external string length limitations.

Reply:
That would be fine with me. Most of the functionality is very specific to SageMaker however, as it's meant to reduce length of the training job name. The only truly generic part is the construction of `suffix` i.e. `suffix = "".join(random.choices(string.ascii_lowercase, k=4))`. I can move that to a string util (and try write a unit test). Anything else you'd consider generic and suitable for moving?

Reply:
https://github.com/zenml-io/zenml/pull/1505/commits/4e1fe2441ede2e85bf001cc300fb1fc02541b845

Reply:
I was thinking a function that takes a string, max_length and a length of random chars might be the reusable part, but you're right outside that specific little piece, everything else is quite Sagemaker-specific.